shared/util/codeql/suppression/AlertSuppression.qll

```diff
@@ -72,6 +78,47 @@ module Make<SingleLineComment Comment> {
     ) {
       this.hasLocationInfo(filepath, startline, _, endline, endcolumn) and
       startcolumn = 1
+      or
+      exists(int cStartLine, int cStartColumn, int cEndLine, int cEndColumn |
+        this.hasLocationInfo(filepath, cStartLine, cStartColumn, cEndLine, cEndColumn) and
+        not exists(int c, Node n | c < cStartColumn |
+          n.hasLocationInfo(filepath, _, _, cStartLine, c) or
+          n.hasLocationInfo(filepath, cStartLine, c, _, _)
+        ) and
+        // when there is no column information, a location spans the whole line
+        startcolumn = 0 and
+        endcolumn = 0 and
+        startline = cEndLine + 1 and
+        endline = startline
+      )
+    }
+  }
+
+  private class CodeQlSuppressionComment extends SuppressionComment {
+    private string annotation;
+
+    CodeQlSuppressionComment() {
+      // match `codeql[...]` anywhere in the comment
+      annotation = this.(Comment).getText().regexpFind("(?i)\\bcodeql\\s*\\[[^\\]]*\\]", _, _) and
```

Comment:
```suggestion
      annotation = this.getText().regexpFind("(?i)\\bcodeql\\s*\\[[^\\]]*\\]", _, _) and
```

Redundant cast?

Reply:
It looked redundant to me too, but the results are empty without it. @alexet do you know why that is?